### PR TITLE
Move code for getting list of custom bootstrap actions to util.tsx

### PR DESCRIPTION
**Pull request by rkilpadi**
## Description

Details.tsx has some code that checks whether Slurm Accounting is active on a cluster. I moved this to utils so that it can be reused to check for other custom bootstrap actions (specifically Slurm REST API).

## How Has This Been Tested?

- Checked that the accounting tab was visible only when an accounting cluster was selected.
- Created tests for `getScripts()`

### CODE REVIEW

File: frontend/src/old-pages/Clusters/util.tsx [hunk 1]
```diff
@@ -27,3 +28,23 @@ export function selectCluster(clusterName: any, navigate: any)
   if(oldClusterName !== clusterName)
     setState(['app', 'clusters', 'selected'], name);
 }
+
+export function getScripts(customActions: [key: string]) {
```
Comment (mendaomn) [inline frontend/src/old-pages/Clusters/util.tsx:32]: Is it possibile that you meant the following?

```ts
customActions: {[key: string]}
```

File: frontend/src/old-pages/Clusters/util.tsx [hunk 2]
```diff
@@ -27,3 +28,23 @@ export function selectCluster(clusterName: any, navigate: any)
...
+    if(getIn(customActions, [actionName, 'Script']))
+      allScripts.push(scriptName(getIn(customActions, [actionName, 'Script'])));
+    for(let arg of (getIn(customActions, [actionName, 'Args']) || []))
+    {
+      if(arg.length > 0 && arg[0] !== '-')
+        allScripts.push(scriptName(arg));
+    }
+  }
+  return allScripts
+}
```
Comment (mendaomn) [inline frontend/src/old-pages/Clusters/util.tsx:50]: This would a be a great opportunity to write a test for this function! Feel free to ping for help if needed!

File: frontend/src/old-pages/Clusters/__tests__/util.test.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { getScripts } from "../util";
+
+describe("given custom actions", () => {
+    describe("when scripts are present", () => {
+        it("should return the script names", () => {
```
Comment (mendaomn) [inline frontend/src/old-pages/Clusters/__tests__/util.test.tsx:5]: suggestion: this doesn't fully capture what the function does. (I know I sent you these lines, let's find something that can document the function)

Here's what I would do:

1. split into two different `it` tests, one per supported custom action. Meaning, one for the `OnNodeStart` and one for the `OnNodeConfigured` since those are the only ones we actually support
2. explicit what this means in terms of business logic, rather than implementation. For instance, you could write `given a custom action to be run before any node deployment, when a custom script is provided, it should return the name of the script`, same goes for the `OnNodeConfigured`
3. test corner cases: `when no custom script is provided` or `when the script should be run with args`

**Comment by mendaomn** [comment]
Great job, LGTM

**Review by mendaomn — APPROVED** [review]
